packages/react-aria-components/src/ListBox.tsx

```diff
@@ -121,8 +117,13 @@ interface ListBoxInnerProps<T> {
   listBoxRef: RefObject<HTMLDivElement | null>
 }
 
-function ListBoxInner<T extends object>({state, props, listBoxRef}: ListBoxInnerProps<T>) {
+function ListBoxInner<T extends object>({state: inputState, props, listBoxRef}: ListBoxInnerProps<T>) {
+  let {filterFn, collectionProps, collectionRef} = useContext(UNSTABLE_InternalAutocompleteContext) || {};
+  props = useMemo(() => collectionProps ? ({...props, ...collectionProps}) : props, [props, collectionProps]);
   let {dragAndDropHooks, layout = 'stack', orientation = 'vertical'} = props;
+  // Memoed so that useAutocomplete callback ref is properly only called once on mount and not everytime a rerender happens
+  listBoxRef = useObjectRef(useMemo(() => mergeRefs(listBoxRef, collectionRef !== undefined ? collectionRef as RefObject<HTMLDivElement> : null), [collectionRef, listBoxRef]));
```

Comment:
given that ref's shouldn't change, could we use `memo` inside `mergeRefs`? then we wouldn't need to worry about this in other places either

Reply:
well we'd need to make a new hook for it since `mergeRefs` is not a hook already. this isn't new though, so not going to do it in this PR.